src/components/Security/Roles/Filters.vue

```diff
@@ -29,27 +40,51 @@
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
+
 export default {
   name: 'RolesFilters',
   components: {},
   props: {
     currentFilter: Object
   },
+  computed: {
+    ...mapGetters('kuzzle', ['$kuzzle'])
+  },
   data() {
     return {
-      controllers: []
+      controllers: [],
+      availableControllers: []
     }
   },
   methods: {
     resetSearch() {
       this.controllers = []
+    },
+    addControllerTag(controller) {
+      if (this.controllers.includes(controller)) {
+        return
+      }
+      this.controllers.push(controller)
+    },
+    async getKuzzlePublicApi() {
+      try {
+        const publicApi = await this.$kuzzle.query({
+          controller: 'server',
+          action: 'publicApi'
+        })
+        this.availableControllers = Object.keys(publicApi.result)
+      } catch (error) {
+        this.$log.error(error)
```

Comment:
It would be nice to warn the user that the controller list is not available, in addition to the log. I'd disable the dropdown and put a tooltip saying "Unable to retrieve controller list"